### PR TITLE
contributing: Adopt the DCO

### PR DESCRIPTION
**Pull request by russellb**
This commit includes a light-weight approach for MetalLB to adopt the
DCO (Developer Certificate of Origin) for contributions.  Some projects
use a `Signed-off-by:` header in commit messages as an acknowledgement
of the DCO.  In this case, we just include the DCO file and update the
contributing docs to explain that by contributing, you agree to the DCO.

Some examples of other projects that adopted this approach to using the
DCO include:

 - several from github.com/coreos, such as github.com/coreos/ignition
 - projects under github.com/metal3-io/

### CODE REVIEW

**Comment by champtar** [comment]
A pretty good tool to enforce `Signed-off-by` if wanted: https://github.com/apps/dco

**Comment by rainleander** [comment]
Thanks so much for initiating this tool and let me know how I can help. If you'd like to read more about what we're asking of our own repositories, please visit https://github.com/packethost/standards

File: CONTRIBUTING.md
Comment (rata) [inline CONTRIBUTING.md:11]: Just curious: this link works because it is in the same directory? Don't we need a `./` nor anything?

**Review by rata — APPROVED** [review]
LGTM, thanks!

Will wait for others to review, just in case :)

File: CONTRIBUTING.md
Comment (russellb) [reply CONTRIBUTING.md:11]: yeah, same directory.  It works on github, at least! Check out: https://github.com/russellb/metallb/blob/dco/CONTRIBUTING.md

**Review by daxmc99 — APPROVED** [review]

**Review by johananl — APPROVED** [review]
LGTM. Will merge soon if there are no concerns.

File: CONTRIBUTING.md
Comment (rata) [reply CONTRIBUTING.md:11]: Thanks!

**Comment by vbatts** [comment]
LGTM

**Comment by johananl** [comment]
Updated the website.

**Pull request merged**
